docs/reference/aggregations/bucket/ipprefix-aggregation.asciidoc

```diff
@@ -0,0 +1,334 @@
+{"ipv4":"192.168.3.107","ipv6":"2001:db8:a4f8:114f:6001:0:12:7307"}
+----------------------------------------------
+// TESTSETUP
+
+The following aggregation will aggregate documents in buckets, each identifying a different sub-network. The sub-network is calculated applying a netmask with prefix length of `24` to each IP address in the `ipv4` field:
+
+[source,console,id=ip-prefix-ipv4-example]
+--------------------------------------------------
+GET /network-traffic/_search
+{
+  "size": 0,
+  "aggs": {
+    "ipv4-subnets": {
+      "ip_prefix": {
+        "field": "ipv4",
+        "prefix_len": 24
+      }
+    }
+  }
+}
+--------------------------------------------------
+// TEST
+
+Response:
+
+[source,console-result]
+--------------------------------------------------
+{
+  ...
+
+  "aggregations": {
+    "ipv4-subnets": {
+      "buckets": [
+        {
+          "key": "192.168.1.0",
+          "is_ipv6": false,
+          "doc_count": 4,
+          "prefix_len": 24,
+          "netmask": "255.255.255.0"
+        },
+        {
+          "key": "192.168.2.0",
+          "is_ipv6": false,
+          "doc_count": 3,
+          "prefix_len": 24,
+          "netmask": "255.255.255.0"
+        },
+        {
+           "key": "192.168.3.0",
+           "is_ipv6": false,
+           "doc_count": 2,
+           "prefix_len": 24,
+           "netmask": "255.255.255.0"
+        }
+      ]
+    }
+  }
+}
+--------------------------------------------------
+// TESTRESPONSE[s/\.\.\./"took": $body.took,"timed_out": false,"_shards": $body._shards,"hits": $body.hits,/]
+
+In a similar way we can aggregate IPv6 addresses, making sure we set the `is_ipv6` flag to `true` (see section below):
+
+[source,console,id=ip-prefix-ipv6-example]
+--------------------------------------------------
+GET /network-traffic/_search
+{
+  "size": 0,
+  "aggs": {
+    "ipv6-subnets": {
+      "ip_prefix": {
+        "field": "ipv6",
+        "prefix_len": 64,
+        "is_ipv6": true
+      }
+    }
+  }
+}
+--------------------------------------------------
+// TEST
+
+Response:
+
+[source,console-result]
+--------------------------------------------------
+{
+  ...
+
+  "aggregations": {
+    "ipv6-subnets": {
+      "buckets": [
+        {
+          "key": "2001:db8:a4f8:112a::",
+          "is_ipv6": true,
+          "doc_count": 4,
+          "prefix_len": 64
+        },
+        {
+          "key": "2001:db8:a4f8:112c::",
+          "is_ipv6": true,
+          "doc_count": 3,
+          "prefix_len": 64
+        },
+        {
+          "key": "2001:db8:a4f8:114f::",
+          "is_ipv6": true,
+          "doc_count": 2,
+          "prefix_len": 64
+        }
+      ]
+    }
+  }
+}
+--------------------------------------------------
+// TESTRESPONSE[s/\.\.\./"took": $body.took,"timed_out": false,"_shards": $body._shards,"hits": $body.hits,/]
+
+NOTE: the `netmask` field is not returned in the response when flag `is_ipv6` is set to `true`.
+
+==== Keyed Response
```

Comment:
If possible, try to set explicit anchor IDs for each heading. Otherwise, our docs build will auto-generate it, which can break the anchors and create namespacing issues.
```suggestion
[[ipprefix-agg-keyed-response]]
==== Keyed Response
```